rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -438,7 +438,14 @@ void MCAPStorage::read_metadata()
     // Look up the offered_qos_profiles metadata entry
     const auto metadata_it = channel.metadata.find("offered_qos_profiles");
     if (metadata_it != channel.metadata.end()) {
-      topic_info.topic_metadata.offered_qos_profiles = metadata_it->second;
+      if (metadata_it->second != "") {
```

Comment:
Here, a conversion from serialized YAML (`std::string`) to `std::vector<rclcpp::QoS>` is needed.